### PR TITLE
[Issue #1659] Create general purpose s3 bucket

### LINKED ISSUE #1659: [Task]: Populate the Search API with seeded data in dev
### Summary

Populate API with factory seeded data (similar to how we can do it locally)

This work is blocking Usability Testing, and frontend connecting to dev and showing data. Staging will be done later

### Acceptance criteria

- API is seeded with mock data for dev

### PR DESCRIPTION
**Pull request by coilysiren**
## Summary

Fixes #1659

## Changes proposed

Adds a "General Purpose" s3 bucket to all of our services. The s3 bucket exposes its location via the `S3_BUCKET_ARN` environment variable. I assume this is going to be a long lived general purpose s3 bucket for... random stuff. Whatever you might want an S3 bucket for. It's also specifically for populating seed data, eg. https://github.com/HHS/simpler-grants-gov/issues/1659

## Context for reviewers

The S3 bucket itself is a near exact copy of the s3 bucket that we use for access logs.

## Additional information

Note that the s3 bucket arn format is like so: `arn:aws:s3:::api-dev-general-purpose20240416221336163000000001`. You probably want to `.split(":")[-1]` it to get just the name.

### CODE REVIEW

**Comment by coilysiren** [comment]
If this looks good, then I guess I can also update the architecture diagram.

**Comment by coilysiren** [comment]
It's not as if the architecture diagram currently lists _**every single**_ s3 bucket though...

**Review by jamesbursa — DISMISSED** [review]

File: infra/modules/service/s3.tf [hunk 1]
```diff
@@ -0,0 +1,102 @@
+resource "aws_s3_bucket" "general_purpose" {
+  bucket_prefix = "${var.service_name}-general-purpose"
+  force_destroy = false
+  # checkov:skip=CKV2_AWS_62:Event notification not necessary for this bucket expecially due to likely use of lifecycle rules
```
[the remark this answers is not in the document]
Comment (coilysiren) [reply infra/modules/service/s3.tf]: ✅ https://github.com/HHS/simpler-grants-gov/pull/1762/commits/8c7e619565280d7ac5d7c93e1772c2777d7660dc

File: infra/modules/service/s3.tf [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    abort_incomplete_multipart_upload {
+      days_after_initiation = 7
+    }
+  }
+
+  rule {
+    id     = "StorageClass"
+    status = "Enabled"
+    dynamic "transition" {
+      for_each = local.log_file_transition
```
[the remark this answers is not in the document]
Comment (coilysiren) [reply infra/modules/service/s3.tf:70]: Not particularly!

File: infra/modules/service/s3.tf [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      }
+    }
+  }
+
+  rule {
+    id     = "Expiration"
+    status = "Enabled"
+    expiration {
+      days = 2555
+    }
```
[the remark this answers is not in the document]
Comment (coilysiren) [reply infra/modules/service/s3.tf:83]: Good catch 👍🏼

File: infra/modules/service/s3.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (coilysiren) [reply infra/modules/service/s3.tf]: https://github.com/HHS/simpler-grants-gov/pull/1762/commits/7d719dc5b659364addf75cef4207a3aa899c453d ✅

**Review by aplybeah — APPROVED** [review]

**Pull request merged**